src/pages/api/repositories/baseRepository.ts

```diff
@@ -0,0 +1,28 @@
+async function handleAsync<T>(operation: () => Promise<T>): Promise<T> {
+  try {
+      return await operation();
+  } catch (error) {
+      console.error("Error during operation:", error);
+      throw error;
+  }
+}
+
+export abstract class BaseRepository<T> {
```

Comment:
Move this file away from api folder. Everything inside that can be called by a client.